src/app/code/community/Omise/Gateway/Controller/Base.php

```diff
@@ -109,9 +109,12 @@ protected function checkPaymentStatusAndUpdateOrder($charge, $order) {
      * If payment is awaiting to capture then updating order status as 'payment_review'. In case of 3-D secured payment,
      * it will be 'processing'.
      * @param Omise_Gateway_Model_Order $order
+     * @param Omise_Gateway_Model_Api_Charge $charge
      * @return Mage_Core_Controller_Varien_Action
      */
-    protected function paymentAwaiting($order) {
+    protected function paymentAwaiting($order, $charge) {
+        if($charge->isUnauthorized() && $order->getStatus() != Mage_Sales_Model_Order::STATE_CANCELED)
```

Comment:
It is often unauthorized. We can check status of payment many times, before we decide to restore cart.
Solution could be for user to decide by himself if he want to restore cart
Displaying `Reorder` button could be solution for that.